src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,561 @@
+        expect(player.getPlaybackRate()).to.equal(4);
+
+        expect(player.setPlaybackRate(3)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(3);
+
+        expect(player.setPlaybackRate(2)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(2);
+
+        expect(player.setPlaybackRate(1.5)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(1.5);
+
+        expect(player.setPlaybackRate(0.7)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(0.7);
+
+        expect(player.setPlaybackRate(1)).to.equal(undefined);
+        expect(player.getPlaybackRate()).to.equal(1);
+      });
+    });
+
+    describe("seekTo", () => {
+      it("should throw in seekTo by default", () => {
+        expect(() => player.seekTo(10)).to.throw();
+        expect(() => player.seekTo(54)).to.throw();
+        expect(() => player.seekTo({ relative: 5 })).to.throw();
+        expect(() => player.seekTo({ position: 5 })).to.throw();
+        expect(() => player.seekTo({ wallClockTime: 5 })).to.throw();
+      });
+    });
+
+    describe("exitFullscreen", () => {
+      it("should allow exitFullscreen by default", () => {
+        expect(player.exitFullscreen()).to.equal(undefined);
+      });
+    });
+
+    describe("setFullscreen", () => {
+      it("should allow setFullscreen by default", () => {
+        expect(player.setFullscreen()).to.equal(undefined);
+
+        // TODO remove for v3.0.0
+        expect(player.setFullscreen(false)).to.equal(undefined);
+      });
+    });
+
+    describe("getVolume/setVolume", () => {
+      it("should throw in setVolume by default if no volume has been given", () => {
+        expect(() => player.setVolume(5)).to.throw();
+      });
+
+      /* tslint:disable:max-line-length */
+      it("should set the volume in setVolume by default if a volume has been given", () => {
+      /* tslint:enable:max-line-length */
+        const videoElement = player.getVideoElement();
+        if (videoElement == null) {
+          throw new Error("The API is disposed");
+        }
+        expect(player.setVolume(1)).to.equal(undefined);
+        expect(player.setVolume(0.5)).to.equal(undefined);
+        expect(player.getVolume()).to.equal(0.5);
+        expect(videoElement.volume).to.equal(0.5);
+
+        expect(player.setVolume(1)).to.equal(undefined);
+        expect(player.getVolume()).to.equal(1);
+        expect(videoElement.volume).to.equal(1);
+      });
+    });
+
+    describe("mute/unMute/isMute", () => {
+      it("should set the volume to 0 in mute by default", () => {
+        const videoElement = player.getVideoElement();
+        if (videoElement == null) {
+          throw new Error("The API is disposed");
+        }
+        if (videoElement.muted) {
+          videoElement.muted = false;
+        }
+        player.setVolume(1);
+
+        expect(player.mute()).to.equal(undefined);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this use of the output from "mute"; "mute" doesn't return anything. ([typescript:S3699](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3699))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=436&issues=AWgOFkKwUMkE2J58zqel&open=AWgOFkKwUMkE2J58zqel)

[ISSUE_KEY]: AWgOFkKwUMkE2J58zqel